general_considerations.md

```diff
@@ -272,6 +379,24 @@ pool to support private contract derivative instruments are:
    the previous epoch)
 3. The ability transact shares across two difficulty adjustment windows.
 
+[kp]
+
+I think we are not addressing how the above three points will be fulfilled. At
+the moment, we are saying we will have a transaction system. We don't talk about
+how the scripting language.
+
+Another concern with introducing a transaction system is that we have make sure
+we safeguard against double spends. Could an attacker with enough hashrate to
+take on braidpool stall progress by constantly thrashing between forks? They
+don't even have to wait for a block to be found. They could keep changing the
+tip so that braidpool can't find a stable enough period to make progress.
+
+We should think this through. Does introducing a transaction system open
+braidpool to more types of attacks than without it? Instinctively, with a larger
+surface area, we will be vulnerable to more types of attacks.
+
+[/kp]
+
```

Comment:
This is a 51% attack, and yes it's possible. I like to call such DAG solutions "ladder" diagrams because the simplest diagram looks like a ladder, with the tip switching sides of the ladder.

This is also controlled by the difficulty adjustment algorithm (DAA). A "ladder" is a single cohort, so the DAA sees no graph cuts for a long time and adjusts the difficulty upwards (beads become slower/difficulty increases/weak block target decreases)